micrometer-registry-signalfx/src/main/java/io/micrometer/signalfx/SignalfxTimer.java

```diff
@@ -62,12 +63,15 @@ final class SignalfxTimer extends AbstractTimer {
 
     private final TimeWindowMax max;
 
+    private final DeltaHistogramSnapshot deltaHistogramSnapshot;
+
     SignalfxTimer(Id id, Clock clock, DistributionStatisticConfig distributionStatisticConfig,
-            PauseDetector pauseDetector, TimeUnit baseTimeUnit, long stepMillis) {
+            PauseDetector pauseDetector, TimeUnit baseTimeUnit, long stepMillis, boolean isDelta) {
         super(id, clock, CumulativeHistogramConfigUtil.updateConfig(distributionStatisticConfig), pauseDetector,
                 baseTimeUnit, false);
         countTotal = new StepTuple2<>(clock, stepMillis, 0L, 0L, count::sumThenReset, total::sumThenReset);
         max = new TimeWindowMax(clock, distributionStatisticConfig);
+        deltaHistogramSnapshot = new DeltaHistogramSnapshot(isDelta);
```

Comment:
Better DI practice is to pass in the `deltaHistogramSnapshot` instance rather than doing the creation here.

Reply:
I agree, but the whole code in micrometer is written this way, look at the `DistributionStatisticConfig` which passes the config and the class receiving the configuration creates the necessary objects. Will be an anti pattern when trying to upstream the code.

Reply:
Yeah ok. I guess we'll favor consistency then.